tests/ha/remove_node.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Remove a node both by its hostname and ip address
+# Maintainer: Julien Adamek <jadamek@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use hacluster;
+
+sub run {
+    my $cluster_name = get_cluster_name;
+    my $node_01      = choose_node(1);
+    my $node_02      = choose_node(2);
+    my $node_01_ip   = get_ip($node_01);
+    my $node_02_ip   = get_ip($node_02);
+    my $remove_cmd   = 'crm cluster remove -y -c';
+    my $join_cmd     = 'crm cluster join -y -w /dev/watchdog -i ' . get_var('SUT_NETDEVICE', 'eth0') . ' -c';
+    my $timer        = 5 * get_var('TIMEOUT_SCALE', 1);
+
+    # Waiting for the other nodes to be ready
+    barrier_wait("REMOVE_NODE_BY_HOST_INIT_$cluster_name");
+
+    # Remove the second node by its hostname
+    assert_script_run "$remove_cmd $node_02" if is_node(1);
+    # Need to wait a bit for cluster configuration refresh
+    sleep $timer;
+
+    # Synchronize all the nodes and save cluster status
+    barrier_wait("REMOVE_NODE_BY_HOST_DONE_$cluster_name");
+    save_state if is_node(1);
```

Comment:
I think it would be useful to see in all nodes that the node has been removed. I would change the above `if is_node(1)` for an `unless is_node(2)` and add some command for node 2 to show that it is no longer a part of the cluster (cannot be `save_state` as it would make the test fail when run on a machine that's not a member of a cluster ... perhaps `script_run $crm_mon_cmd` as `$crm_mon_cmd` is already exported by `lib/hacluster` ???)

Reply:
Totally agree, thanks for the reminder about 'unless'.
Indeed, $crm_mon_cmd is already exported by lib/hacluster.